nixos/modules/services/networking/nat.nix

```diff
@@ -55,6 +55,16 @@ let
         -j DNAT --to-destination ${fwd.destination}
     '') cfg.forwardPorts}
 
+    # Allow to connect to ${cfg.externalIP}:${fwd.sourcePort} from the host itself
+    ${optionalString (cfg.externalIP != null) (
```

Comment:
Maybe we could make another rule for all traffic coming from the loopback interface, so we got that covered even when `externalIP` isn't set? This solves the "connectable from the Host itself" for traffic in the usual ranges, e.g. `127.0.0.0/8`.

Reply:
I am not sure that loopback is involved here, source IP is ```eth0```'s, not ```127.0.0.1``` and it works without ```net.ipv4.conf.all.route_localnet``` 

Anyway, I agree that ```cfg.externalIP``` meaning is overloaded.

Reply:
Connectivity to```127.0.0.1:${fwd.sourcePort}``` is another issue, closer to @mogorman's